packages/augur-ui/src/modules/modal/withdraw-form.tsx

```diff
@@ -74,35 +87,63 @@ export class WithdrawForm extends Component<
   };
 
   handleMax = () => {
-    const { loginAccount, GasCosts } = this.props;
+    const { loginAccount, fallBackGasCosts } = this.props;
     const { currency } = this.state;
     const fullAmount = createBigNumber(
       loginAccount.balances[currency.toLowerCase()]
     );
-    const valueMinusGas = fullAmount.minus(GasCosts.eth.fullPrecision);
+    const valueMinusGas = fullAmount.minus(fallBackGasCosts.eth.fullPrecision);
     const resolvedValue = valueMinusGas.lt(ZERO) ? ZERO : valueMinusGas;
     this.amountChange(
       currency === ETH ? resolvedValue.toFixed() : fullAmount.toFixed()
     );
   };
 
   amountChange = (amount: string) => {
-    const { loginAccount, GasCosts } = this.props;
+    const {
+      loginAccount,
+      fallBackGasCosts,
+      Gnosis_ENABLED,
+      ethToDaiRate,
+      gasPrice,
+    } = this.props;
+    const { relayerGasCosts } = this.state;
     const newAmount = convertExponentialToDecimal(sanitizeArg(amount));
     const bnNewAmount = createBigNumber(newAmount || '0');
     const { errors: updatedErrors, currency } = this.state;
     updatedErrors.amount = '';
     const availableEth = createBigNumber(loginAccount.balances.eth);
+    const availableDai = createBigNumber(loginAccount.balances.dai);
     let amountMinusGas = ZERO;
-    if (currency === ETH && newAmount) {
-      amountMinusGas = availableEth
-        .minus(bnNewAmount)
-        .minus(GasCosts.eth.fullPrecision);
+
+    if (Gnosis_ENABLED) {
+      const relayerGasCostsETH = formatGasCostToEther(
+        relayerGasCosts,
+        { decimalsRounded: 4 },
+        gasPrice
+      );
+      const relayerGasCostsDAI = ethToDai(relayerGasCostsETH, ethToDaiRate);
+
+      if (currency === DAI) {
+        amountMinusGas = availableDai
+          .minus(bnNewAmount)
+          .minus(createBigNumber(relayerGasCostsDAI.value));
+      } else {
+        amountMinusGas = availableDai.minus(
+          createBigNumber(relayerGasCostsDAI.value)
+        );
+      }
     } else {
-      amountMinusGas =
-        currency === DAI
-          ? availableEth.minus(GasCosts.dai.fullPrecision)
-          : availableEth.minus(GasCosts.rep.fullPrecision);
+      if (currency === ETH && newAmount) {
+        amountMinusGas = availableEth
+          .minus(bnNewAmount)
+          .minus(fallBackGasCosts.eth.fullPrecision);
+      } else {
+        amountMinusGas =
+          currency === DAI
+            ? availableEth.minus(fallBackGasCosts.dai.fullPrecision)
```

Comment:
Why subtract from available ETH balance?

Reply:
Good eye, that's a mistake!

Reply:
actually, this is fine. 

We subtract from available ETH balance b/c we want to determine how much ETH is remaining once they pay the gas costs to send DAI.

Reply:
`fallBackGasCosts.dai.fullPrecision` is this actually ETH transaction fee?